File: C#/PokerProtocol/Commands/Lobby/Response/ListTableResponse.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Lobby.Response
{
    public class ListTableResponse : AbstractLobbyCommandResponse<ListTableCommand>
    {

        protected override string CommandName
        {
            get { return COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyLIST_TABLES_RESPONSE";
        private readonly List<TupleTableInfo> m_Tables;
        public List<TupleTableInfo> Tables
        {
            get { return m_Tables; }
        }


        public ListTableResponse(StringTokenizer argsToken)
            : base(new ListTableCommand(argsToken))
        {
            m_Tables = new List<TupleTableInfo>();
            int count = int.Parse(argsToken.NextToken());
            for (int i = 0; i < count; ++i)
            {
                m_Tables.Add(new TupleTableInfo(argsToken));
            }
        }

        public ListTableResponse(ListTableCommand command, List<TupleTableInfo> tables)
            : base(command)
        {
            m_Tables = tables;
        }

        public override void Encode(StringBuilder sb)
        {
            base.Encode(sb);
            Append(sb, m_Tables.Count);
            foreach (TupleTableInfo tti in m_Tables)
                Append(sb, tti.ToString(AbstractLobbyCommand.Delimitter));
        }
    }
}

File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/lobby/ListTableResponse.java
package bluffinmuffin.protocol.commands.lobby;

import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

import bluffinmuffin.protocol.TupleTableInfo;

public class ListTableResponse extends AbstractLobbyResponse<ListTableCommand>
{
    @Override
    protected String getCommandName()
    {
        return ListTableResponse.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyLIST_TABLES_RESPONSE";
    private final List<TupleTableInfo> m_tables;
    
    public ListTableResponse(StringTokenizer argsToken)
    {
        super(new ListTableCommand(argsToken));
        m_tables = new ArrayList<TupleTableInfo>();
        final int count = Integer.parseInt(argsToken.nextToken());
        for (int i = 0; i < count; ++i)
        {
            m_tables.add(new TupleTableInfo(argsToken));
        }
    }
    
    public ListTableResponse(ListTableCommand command, List<TupleTableInfo> tables)
    {
        super(command);
        m_tables = tables;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        super.encode(sb);
        append(sb, m_tables.size());
        for (final TupleTableInfo info : m_tables)
        {
            append(sb, info.toString("" + AbstractLobbyCommand.Delimitter));
        }
    }
    
    public List<TupleTableInfo> getTables()
    {
        return m_tables;
    }
}
